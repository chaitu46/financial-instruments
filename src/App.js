import useSampleData from "./hooks/useSampleData";
import { StyledAppTitle, StyledTableContainer } from "./styledComponents";
import { Table } from "./Table";

export const App = () => {
  const sampleData = useSampleData(); // getting sample from custom hook, So custom hook can be leveraged for API call fetch in future.
  return (
    <div>
      <StyledAppTitle>Financial Instruments - Assessment</StyledAppTitle>
      <StyledTableContainer>
        <Table initialData={sampleData} />
      </StyledTableContainer>
    </div>
  );
};

export default App;
