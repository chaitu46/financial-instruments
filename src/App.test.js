import { render, screen } from "@testing-library/react";
import App from "./App";

jest.mock("./hooks/useSampleData", () => () => testData);

const testData = [
  {
    ticker: "BETA",
    price: 3791.37,
    assetClass: "Equities",
  },
  {
    ticker: "EPSILON",
    price: 1168.46,
    assetClass: "Credit",
  },
  {
    ticker: "IOTA",
    price: -1096.64,
    assetClass: "Macro",
  },
  {
    ticker: "GAMMA",
    price: 2299.1,
    assetClass: "Equities",
  },
];

describe("App", () => {
  test("renders default app", () => {
    render(<App />);
    const pageHeading = screen.getByRole("heading");
    const table = screen.getByRole("table");

    expect(pageHeading).toHaveTextContent("Financial Instruments - Assessment");
    expect(table).toBeVisible();
    expect(table).toHaveAttribute("aria-label", "Financial Details");
    expect(table).toMatchSnapshot();
  });
});
