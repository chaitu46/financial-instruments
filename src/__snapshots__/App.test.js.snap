// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders default app 1`] = `
<table
  aria-label="Financial Details"
  class="sc-bdfBwQ keewPp"
>
  <thead>
    <tr
      class="sc-hKgILt eBPHrI"
    >
      <th>
        <button
          class="sc-eCssSg gYaMmc"
          data-active="false"
          data-testid="assertClass"
          id="assertClass"
        >
          Asset Class
        </button>
      </th>
      <th>
        <button
          class="sc-eCssSg gYaMmc"
          data-active="false"
          data-testid="ticker"
          id="ticker"
        >
          Ticker
        </button>
      </th>
      <th>
        <button
          class="sc-eCssSg gYaMmc"
          data-active="false"
          data-testid="price"
          id="price"
        >
          Price
        </button>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      class="sc-gsTCUz kHDUxD"
      data-theme="[object Object]"
    >
      <td>
        Equities
      </td>
      <td>
        BETA
      </td>
      <td
        class="sc-dlfnbm eWIIxE"
        data-bg="#1d4289"
      >
        3791.37
      </td>
    </tr>
    <tr
      class="sc-gsTCUz ibRWgQ"
      data-theme="[object Object]"
    >
      <td>
        Credit
      </td>
      <td>
        EPSILON
      </td>
      <td
        class="sc-dlfnbm eWIIxE"
        data-bg="#1d4289"
      >
        1168.46
      </td>
    </tr>
    <tr
      class="sc-gsTCUz jmeBcN"
      data-theme="[object Object]"
    >
      <td>
        Macro
      </td>
      <td>
        IOTA
      </td>
      <td
        class="sc-dlfnbm kxGDBW"
        data-bg="#d3273e"
      >
        -1096.64
      </td>
    </tr>
    <tr
      class="sc-gsTCUz kHDUxD"
      data-theme="[object Object]"
    >
      <td>
        Equities
      </td>
      <td>
        GAMMA
      </td>
      <td
        class="sc-dlfnbm eWIIxE"
        data-bg="#1d4289"
      >
        2299.1
      </td>
    </tr>
  </tbody>
</table>
`;
